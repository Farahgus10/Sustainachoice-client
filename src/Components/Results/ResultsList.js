import React from 'react';
import { Link } from 'react-router-dom'

function ResultsList(props) {
      const result = (!props.results || props.results.length === 0) ? <div className="no-results"><p >No results</p></div> : 
      props.results.map((item, i) => 
      {
        if (props.expandedView === i ) {
        return (<li key={item.id} onClick={() => props.toggleExpandedItem(-1) }>
        <p className="item-name">{item.location_name}</p>
        <p className="item-info">{item.info}</p>
        <p>{item.location_type}</p>
        <p>{item.location_address}</p>
        <a href={item.website}>{item.website}</a>
        <p></p>
        <Link to={`/location/${item.id}`}>Want to rate this location or have something to say about it? Click here to add a comment!</Link>
      </li>)
      } else {
        return (
           <li key={item.id} onClick={() => props.toggleExpandedItem(i)}>
          <p className="item-name">{item.location_name}</p>
          </li>
        )
      }
    })
  
  return (
    <div className="ResultsList" >
      <div className="results" >
      <ul>
        {result}
      </ul>
      </div>
    </div>
  )
}


export default ResultsList;
