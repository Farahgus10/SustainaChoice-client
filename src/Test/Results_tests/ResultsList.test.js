import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import ResultsList from '../../Components/Results/ResultsList';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const resultsList = [ 
  {
    location_name: "Location name 1",
    location_type: "Locationt 1 type x",
    info: "Location 1 info", 
    location_address: "location 1 address",
    website: "location 1 website"
  },
  {
    location_name: "Location name 2",
    location_type: "Locationt 2 type x",
    info: "Location 2 info", 
    location_address: "location 2 address",
    website: "location 2 website"
  },
]


it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ResultsList />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

// it('renders correctly when there are no results', () => {
//   const tree = renderer.create(<ResultsList />)
//     .toJSON();
//     expect(tree).toMatchSnapshot();
// })

// it('renders correctly when there are mulitples items', () => {
//   const tree = renderer.create(<ResultsList results={resultsList} />)
//     .toJSON();
//     expect(tree).toMatchSnapshot();
// })