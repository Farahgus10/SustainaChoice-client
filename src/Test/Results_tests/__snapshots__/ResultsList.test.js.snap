// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are mulitples items 1`] = `
<div
  className="ResultsList"
>
  <div
    className="results"
  >
    <ul>
      <li
        onClick={[Function]}
      >
        <p>
          Location name 1
        </p>
      </li>
      <li
        onClick={[Function]}
      >
        <p>
          Location name 2
        </p>
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders correctly when there are no results 1`] = `
<div
  className="ResultsList"
>
  <div
    className="results"
  >
    <ul>
      <p>
        No results
      </p>
    </ul>
  </div>
</div>
`;
